### PR TITLE
dashboard, p2p, vendor: visualize peers

### PR DESCRIPTION
**Pull request by kurkomisi**
Cleaner version of https://github.com/ethereum/go-ethereum/pull/17911

### CODE REVIEW

File: dashboard/peers.go [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+// removed from the beginning of its array.
+//
+// This data structure makes it possible to marshal the peer
+// history simply by passing it to the JSON marshaler.
+type peerContainer struct {
+	// Bundles is the outer map using the peer's IP address as key.
+	Bundles map[string]*peerBundle `json:"bundles,omitempty"`
+
+	// activePeers contains the peers with opened connection in random order.
+	activePeers *list.List
```
Comment (karalabe) [inline dashboard/peers.go]: Could we use slices instead? Perhaps lets investigate either here or in a followup pr.

File: dashboard/peers.go [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+type peerAttempt struct {
+	// Connected contains the timestamp of the connection attempt's moment.
+	Connected time.Time `json:"connected"`
+
+	// Disconnected contains the timestamp of the
+	// moment when the connection attempt failed.
+	Disconnected time.Time `json:"disconnected"`
+}
+
+type RemovedPeerType string
```
Comment (karalabe) [inline dashboard/peers.go]: dox

**Comment by karalabe** [comment]
There seems to be some glitch in the UI code:

screenshot from 2019-01-11 10-22-52

Also, I've been looking at the peer list, I don't think it's correct. I currently have 1 persistent peer I'm constantly downloading from, but it's not even on the UI at all.

Tried refreshing, but now the entire page crashes instnatly:

```
bundle.js:15200 TypeError: Cannot read property 'push' of undefined
    at bundle.js:1628
    at Array.forEach (<anonymous>)
    at bundle.js:1594
    at t (bundle.js:529)
    at bundle.js:532
    at Array.forEach (<anonymous>)
    at t (bundle.js:531)
    at Dashboard.<anonymous> (bundle.js:640)
    at ur (bundle.js:14397)
    at cr (bundle.js:14409)
ga @ bundle.js:15200
bundle.js:16023 Uncaught TypeError: Cannot read property 'push' of undefined
    at bundle.js:1628
    at Array.forEach (<anonymous>)
    at bundle.js:1594
    at t (bundle.js:529)
    at bundle.js:532
    at Array.forEach (<anonymous>)
    at t (bundle.js:531)
    at Dashboard.<anonymous> (bundle.js:640)
    at ur (bundle.js:14397)
    at cr (bundle.js:14409)
```

Geth in the mean time is spitting out `WARN [01-11|10:36:04.610] Failed to load the asset                 path=/bundle.js.map err="Asset bundle.js.map not found"`. I'm assuming that's the source map for debuggability? Please add the source map too to the assets.

**Comment by kurkomisi** [comment]
I'm investigating the consistency of the UI too.

**Comment by kurkomisi** [comment]
Extended here https://github.com/ethereum/go-ethereum/pull/19247

**Pull request closed without merging**